### ISSUE TITLE
Edge/IE Issue

### ISSUE THREAD

**Issue by cevenx** [creativetimofficial/material-dashboard-pro-angular#134]
Hello. 

I am running into the following error while trying to use my site in IE11, unfortunately this is the environment many of my users are required to use (this issue does not happen in Chrome)

i am using an image below to show you this is happening on your side as well.
![image](https://user-images.githubusercontent.com/27730639/32557384-fdd3f304-c46f-11e7-9a4f-42557c61fd70.png)

this error pops up in the console and i can not use the nav menu correctly. i am able to have the drop-downs function correctly, but the links are then broken.

i my site the error performs in the same manor, with a broken nav system., but the error is a bit more manageable.

![image](https://user-images.githubusercontent.com/27730639/32557763-111b7620-c471-11e7-9697-bdb02c955bd0.png)

that lines up with the following code:

![image](https://user-images.githubusercontent.com/27730639/32557937-991dfce6-c471-11e7-9855-d43d76c084d1.png)

**Comment by chelaruc** [comment]
Hi @cevenx,

Thank you for using our product! Can you try something like this:
1) Delete 119-126 lines
2) Declare this in app.component.ts:
```
private lastPoppedUrl: string;
private yScrollStack: number[] = [];
```
3) In ngOnInit() add this:
```
const elemMainPanel = <HTMLElement>document.querySelector('.main-panel');
const elemSidebar = <HTMLElement>document.querySelector('.sidebar .sidebar-wrapper');

this.location.subscribe((ev:PopStateEvent) => {
    this.lastPoppedUrl = ev.url;
 });
   this.router.events.subscribe((event:any) => {
     this.navbar.sidebarClose();
     if (event instanceof NavigationStart) {
        if (event.url != this.lastPoppedUrl)
            this.yScrollStack.push(window.scrollY);
        } else if (event instanceof NavigationEnd) {
            if (event.url == this.lastPoppedUrl) {
               this.lastPoppedUrl = undefined;
               window.scrollTo(0, this.yScrollStack.pop());
            } else
              window.scrollTo(0, 0);
    }
  });
    this._router = this.router.events.filter(event => event instanceof NavigationEnd).subscribe((event: NavigationEnd) => {
        elemMainPanel.scrollTop = 0;
        elemSidebar.scrollTop = 0;
     });
```
Tell me if it worked for you!

All the best,
Ciprian

**Comment by cevenx** [comment]
my app.conponents.ts now looks like this:

import { Component, OnInit, Inject } from '@angular/core';
import {Router, NavigationEnd, ActivatedRoute, NavigationStart} from '@angular/router';
import { Subscription } from 'rxjs/Subscription';
import 'rxjs/add/operator/filter';
import { DOCUMENT } from '@angular/platform-browser';
declare var $: any;

@Component({
    selector: 'app-my-app',
    templateUrl: './app.component.html'
})

export class AppComponent implements OnInit {
    private _router: Subscription;
    private lastPoppedUrl: string;
    private yScrollStack: number[] = [];

    constructor( private router: Router, @Inject(DOCUMENT, ) private document: any) {}

    ngOnInit() {
        const elemMainPanel = <HTMLElement>document.querySelector('.main-panel');
        const elemSidebar = <HTMLElement>document.querySelector('.sidebar .sidebar-wrapper');


        $.material.options.autofill = true;
        $.material.init();
        this.location.subscribe((ev:PopStateEvent) => {
            this.lastPoppedUrl = ev.url;
        });
        this.router.events.subscribe((event:any) => {
            this.navbar.sidebarClose();
            if (event instanceof NavigationStart) {
                if (event.url != this.lastPoppedUrl)
                    this.yScrollStack.push(window.scrollY);
            } else if (event instanceof NavigationEnd) {
                if (event.url == this.lastPoppedUrl) {
                    this.lastPoppedUrl = undefined;
                    window.scrollTo(0, this.yScrollStack.pop());
                } else
                    window.scrollTo(0, 0);
            }
        });
        this._router = this.router.events.filter(event => event instanceof NavigationEnd).subscribe((event: NavigationEnd) => {
            elemMainPanel.scrollTop = 0;
            elemSidebar.scrollTop = 0;
        });

    }

and i get the following errors:
ERROR in F:/dev/CMS/Client/src/app/app.component.ts (27,14): Property 'location' does not exist on type 'AppComponent'.
ERROR in F:/dev/CMS/Client/src/app/app.component.ts (28,37): Property 'url' does not exist on type 'PopStateEvent'.
ERROR in F:/dev/CMS/Client/src/app/app.component.ts (31,18): Property 'navbar' does not exist on type 'AppComponent'.

**Comment by chelaruc** [comment]
Add this to imports:
```
import { Location, LocationStrategy, PathLocationStrategy, PopStateEvent } from '@angular/common';
````
Replace the constructor with this:
```
constructor( public location: Location, private router: Router) {}
```
and delete:
```
this.navbar.sidebarClose();
```
Tell me if it worked :)

**Comment by cevenx** [comment]
after applying this i have a new error. the same navigation issue persists

```

DOM7011: The code on this page disabled back and forward caching. For more information, see: http://go.microsoft.com/fwlink/?LinkID=291337
File: dashboard
HTML1300: Navigation occurred.
File: dashboard
HTML1500: Tag cannot be self-closing. Use an explicit closing tag.
File: dashboard, Line: 24, Column: 17
Angular is running in the development mode. Call enableProdMode() to enable the production mode.
ERROR Error: Uncaught (in promise): TypeError: Unable to set property 'scrollTop' of undefined or null reference
TypeError: Unable to set property 'scrollTop' of undefined or null reference
   at Anonymous function (http://localhost:4200/main.bundle.js:144:13)
   at SafeSubscriber.prototype.__tryOrUnsub (http://localhost:4200/vendor.bundle.js:18557:13)
   at SafeSubscriber.prototype.next (http://localhost:4200/vendor.bundle.js:18504:17)
   at Subscriber.prototype._next (http://localhost:4200/vendor.bundle.js:18444:9)
   at Subscriber.prototype.next (http://localhost:4200/vendor.bundle.js:18408:13)
   at FilterSubscriber.prototype._next (http://localhost:4200/vendor.bundle.js:21420:13)
   at Subscriber.prototype.next (http://localhost:4200/vendor.bundle.js:18408:13)
   at Subject.prototype.next (http://localhost:4200/vendor.bundle.js:18152:17)
   at Anonymous function (http://localhost:4200/vendor.bundle.js:136266:17)
   at ZoneDelegate.prototype.invoke (http://localhost:4200/polyfills.bundle.js:6082:13)
   at
   "ERROR"
   {
      [functions]: ,
      __proto__: { },
      description: "Uncaught (in promise): TypeError: Unable to set property 'scrollTop' of undefined or null reference
TypeError: Unable to set property 'scrollTop' of undefined or null reference
   at Anonymous function (http://localhost:4200/main.bundle.js:144:13)
   at SafeSubscriber.prototype.__tryOrUnsub (http://localhost:4200/vendor.bundle.js:18557:13)
   at SafeSubscriber.prototype.next (http://localhost:4200/vendor.bundle.js:18504:17)
   at Subscriber.prototype._next (http://localhost:4200/vendor.bundle.js:18444:9)
   at Subscriber.prototype.next (http://localhost:4200/vendor.bundle.js:18408:13)
   at FilterSubscriber.prototype._next (http://localhost:4200/vendor.bundle.js:21420:13)
   at Subscriber.prototype.next (http://localhost:4200/vendor.bundle.js:18408:13)
   at Subject.prototype.next (http://localhost:4200/vendor.bundle.js:18152:17)
   at Anonymous function (http://localhost:4200/vendor.bundle.js:136266:17)
   at ZoneDelegate.prototype.invoke (http://localhost:4200/polyfills.bundle.js:6082:13)
   at onInvoke (ht",
      message: "Uncaught (in promise): TypeError: Unable to set property 'scrollTop' of undefined or null reference
TypeError: Unable to set property 'scrollTop' of undefined or null reference
   at Anonymous function (http://localhost:4200/main.bundle.js:144:13)
   at SafeSubscriber.prototype.__tryOrUnsub (http://localhost:4200/vendor.bundle.js:18557:13)
   at SafeSubscriber.prototype.next (http://localhost:4200/vendor.bundle.js:18504:17)
   at Subscriber.prototype._next (http://localhost:4200/vendor.bundle.js:18444:9)
   at Subscriber.prototype.next (http://localhost:4200/vendor.bundle.js:18408:13)
   at FilterSubscriber.prototype._next (http://localhost:4200/vendor.bundle.js:21420:13)
   at Subscriber.prototype.next (http://localhost:4200/vendor.bundle.js:18408:13)
   at Subject.prototype.next (http://localhost:4200/vendor.bundle.js:18152:17)
   at Anonymous function (http://localhost:4200/vendor.bundle.js:136266:17)
   at ZoneDelegate.prototype.invoke (http://localhost:4200/polyfills.bundle.js:6082:13)
   at onInvoke (ht",
      name: "Error",
      originalStack: "Error: Uncaught (in promise): TypeError: Unable to set property 'scrollTop' of undefined or null reference
TypeError: Unable to set property 'scrollTop' of undefined or null reference
   at Anonymous function (http://localhost:4200/main.bundle.js:144:13)
   at SafeSubscriber.prototype.__tryOrUnsub (http://localhost:4200/vendor.bundle.js:18557:13)
   at SafeSubscriber.prototype.next (http://localhost:4200/vendor.bundle.js:18504:17)
   at Subscriber.prototype._next (http://localhost:4200/vendor.bundle.js:18444:9)
   at Subscriber.prototype.next (http://localhost:4200/vendor.bundle.js:18408:13)
   at FilterSubscriber.prototype._next (http://localhost:4200/vendor.bundle.js:21420:13)
   at Subscriber.prototype.next (http://localhost:4200/vendor.bundle.js:18408:13)
   at Subject.prototype.next (http://localhost:4200/vendor.bundle.js:18152:17)
   at Anonymous function (http://localhost:4200/vendor.bundle.js:136266:17)
   at ZoneDelegate.prototype.invoke (http://localhost:4200/polyfills.bundle.js:6082:13)
   at onInv",
      promise: { },
      rejection: { },
      stack: "Error: Uncaught (in promise): TypeError: Unable to set property 'scrollTop' of undefined or null reference
TypeError: Unable to set property 'scrollTop' of undefined or null reference
   at Anonymous function (http://localhost:4200/main.bundle.js:144:13)
   at SafeSubscriber.prototype.__tryOrUnsub (http://localhost:4200/vendor.bundle.js:18557:13)
   at SafeSubscriber.prototype.next (http://localhost:4200/vendor.bundle.js:18504:17)
   at Subscriber.prototype._next (http://localhost:4200/vendor.bundle.js:18444:9)
   at Subscriber.prototype.next (http://localhost:4200/vendor.bundle.js:18408:13)
   at FilterSubscriber.prototype._next (http://localhost:4200/vendor.bundle.js:21420:13)
   at Subscriber.prototype.next (http://localhost:4200/vendor.bundle.js:18408:13)
   at Subject.prototype.next (http://localhost:4200/vendor.bundle.js:18152:17)
   at Anonymous function (http://localhost:4200/vendor.bundle.js:136266:17)
   at ZoneDelegate.prototype.invoke (http://localhost:4200/polyfills.bundle.js:6082:13)
   at onInv",
      task: { },
      zone: { },
      zoneAwareStack: "Error: Uncaught (in promise): TypeError: Unable to set property 'scrollTop' of undefined or null reference
TypeError: Unable to set property 'scrollTop' of undefined or null reference
   at Anonymous function (http://localhost:4200/main.bundle.js:144:13)
   at SafeSubscriber.prototype.__tryOrUnsub (http://localhost:4200/vendor.bundle.js:18557:13)
   at SafeSubscriber.prototype.next (http://localhost:4200/vendor.bundle.js:18504:17)
   at Subscriber.prototype._next (http://localhost:4200/vendor.bundle.js:18444:9)
   at Subscriber.prototype.next (http://localhost:4200/vendor.bundle.js:18408:13)
   at FilterSubscriber.prototype._next (http://localhost:4200/vendor.bundle.js:21420:13)
   at Subscriber.prototype.next (http://localhost:4200/vendor.bundle.js:18408:13)
   at Subject.prototype.next (http://localhost:4200/vendor.bundle.js:18152:17)
   at Anonymous function (http://localhost:4200/vendor.bundle.js:136266:17)
   at ZoneDelegate.prototype.invoke (http://localhost:4200/polyfills.bundle.js:6082:13)
   at onInv"
   }


```

**Comment by chelaruc** [comment]
@cevenx Can you give me your project for testing? My email address is ciprian@creative-tim.com

**Comment by cevenx** [comment]
I should be able to send it to you in the next few hours, thank you for taking a look.

**Comment by muryavuz** [comment]
this._router = this.router.events.filter(event => event instanceof NavigationEnd).subscribe((event: NavigationEnd) => {
      const elemMainPanel = <HTMLElement>document.querySelector('.main-panel');
      const elemSidebar = <HTMLElement>document.querySelector('.sidebar .sidebar-wrapper');

      if (elemMainPanel) {
        elemMainPanel.scrollTop = 0;
      }
      if (elemSidebar) {
        elemSidebar.scrollTop = 0;
      }
    });

**Comment by chelaruc** [comment]
fixed in v1.4.6
